doc/classes/TreeItem.xml

```diff
@@ -662,6 +669,14 @@
 				Modulates the given column's icon with [param modulate].
 			</description>
 		</method>
+		<method name="set_icon_overlay">
```

Comment:
I wonder if we should just change the icon instead of exposing another new method in `Tree`.

Reply:
Symlink can be any file or folder with different icons, changing icon would require generating a new icon for each.